File: Assets/ToLua/Lua/user/ui/shop/model/info/shop_gold_item_info.lua
local t = {}
t.__index = t

local gameResData = require('ui/game/model/game_res_data')

function t.New (shopLimitItemData)
  local o = {}
  setmetatable(o, t)

  o.shopLimitItemData = shopLimitItemData
  o.isOpen = false
  o.remainPurchaseTimes = 0
  
  return o
end

function t:SetInfo (isOpen, remainPurchaseTimes)
  self.isOpen = isOpen
  self.remainPurchaseTimes = remainPurchaseTimes
end

function t:GetShopItemType ()               --商店类型ID，对应不同的页签
  return self.shopLimitItemData.sheet_num
end

function t:GetShopItemID ()
  return self.shopLimitItemData.id
end

function t:GetShopItemName ()
  return LocalizationController.instance:Get(self.shopLimitItemData.name)
end

function t:GetCostResData ()
  return gameResData.NewByString(self.shopLimitItemData.cost)
end

function t:GetCostType ()
  return 1
end

return t

File: Assets/Code/Logic/Shop/Model/ShopGoldItemInfo.cs
namespace Logic.Shop.Model
{
	public class ShopGoldItemInfo
	{
		private ShopLimitItemData _shopLimitItemData;
		public ShopLimitItemData ShopLimitItemData
		{
			get
			{
				return _shopLimitItemData;
			}
		}

		private bool _isOpen = false;
		public bool IsOpen
		{
			get
			{
				return _isOpen;
			}
		}

		private int _remainPurchaseTimes = 0;
		public int RemainPurchaseTimes
		{
			get
			{
				return _remainPurchaseTimes;
			}
		}

		public ShopGoldItemInfo (ShopLimitItemData shopLimitItemData)
		{
			_shopLimitItemData = shopLimitItemData;
		}

		public void UpdateInfo (bool isOpen, int remainPurchaseTimes)
		{
			_isOpen = isOpen;
			_remainPurchaseTimes = remainPurchaseTimes;
		}
	}
}
